posts/2024-02-13-24.0.0.2-beta.adoc

```diff
@@ -0,0 +1,173 @@
+
+== Back-channel logout for OpenID Connect clients
+
+With this release, back-channel logout for OpenID clients is enabled by default when the link:{url-prefix}/docs/latest/reference/feature/openidConnectClient.html[OpenID Connect Client] and link:{url-prefix}/docs/latest/reference/feature/socialLogin.html[Social Media Login] features are enabled. The following back-channel logout endpoints are enabled for those features:
+
+* OpenID Connect Client: `/oidcclient/backchannel_logout/{clientId}`
+* Social Media Login: `/ibm/api/social-login/backchannel_logout/{clientId}`
+
+== Back-channel logout for OpenID Connect servers
+
+Two options are available to enable back-channel logout for OpenID Connect servers:
+
+* Use the `backchannel_logout_uri` metadata value to specify the OpenID Connect client's back-channel logout URI when the OAuth client is dynamically registered.
+* Use the `backchannelLogoutUri` attribute for OAuth clients that are defined in a `localStore` element.
+
+Also, the OpenID Connect server waits for a default of 180 seconds before the back-channel logout request times out. You can specify a custom duration by using the `backchannelLogoutRequestTimeout` attribute for the `openidConnectProvider` element.
+
+The following `server.xml` file example demonstrates how to specify the back-channel logout URI that an OpenID Connect server uses to make back-channel logout requests, with a timeout of 60 seconds.
+
+[source,xml]
+----
+<featureManager>
+    <feature>openidConnectServer-1.0</feature>
+</featureManager>
+
+...
+
+<openidConnectProvider
+    id="OidcConfigSample"
+    backchannelLogoutRequestTimeout="60s"
+    oauthProviderRef="OAuthConfigSample" ... />
+
+<oauthProvider id="OAuthConfigSample" ... >
+    <localStore>
+        <client
+            name="client01"
+            backchannelLogoutUri="http://localhost:9080/oidcclient/backchannel_logout/client01"
+            ... />
+    </localStore>
+</oauthProvider>
+----
+
+
+Now, when the OpenID Connect server's logout or `end_session` endpoint is invoked, the OpenID Connect server also sends back-channel logout requests to log out the user at the OpenID Connect clients.
+
+The OpenID Connect server's logout endpoint is `/oidc/endpoint/{oidcProviderId}/logout` and the `end_session` endpoint is `/oidc/endpoint/{oidcProviderId}/end_session`.
+
+Optionally, you can add an `id_token_hint` that contains an ID Token that is issued by that OpenID Connect server to the endpoint request as a query parameter to help determine the user to log out. This configuration is useful in scenarios where the logout or `end_session` request is made without the user's OpenID Connect server SSO cookie. Also, if the OpenID Connect server is configured with a SAML Identity Provider (IdP), a logout at the IdP also triggers sends back-channel logout requests to the OpenID Connect clients.
```

Comment:
- can we show an example? this needs some elaboration
- does token_hint trigger logout on the SAML IdP? or is that unrelated?

Reply:
- separated the logout and end_session part from the SAML part to make the distinction of the two parts more clear
- added examples for using logout and end_session (with and without `id_token_hint`)
- added a bit more info for the SAML part

```
Now, when the OpenID Connect server's logout or `end_session` endpoint is invoked, the OpenID Connect server also sends back-channel logout requests to log out the user at the OpenID Connect clients. The OpenID Connect server's logout endpoint is `/oidc/endpoint/{oidcProviderId}/logout` and the `end_session` endpoint is `/oidc/endpoint/{oidcProviderId}/end_session`. Optionally, you can add an `id_token_hint` that contains an ID Token that is issued by that OpenID Connect server to the endpoint request as a query parameter to help determine the user to log out. This configuration is useful in scenarios where the logout or `end_session` request is made without the user's OpenID Connect server SSO cookie. 

For example, if the OpenID Connect server is hosted on `http://localhost:9081` and the user’s OpenID Connect server’s SSO cookie is available, then invoking either of the following endpoints will cause the OpenID Connect server to also send back-channel logout requests to the configured back-channel logout URI’s.

* Logout endpoint: `http://localhost:9081/oidc/endpoint/{oidcProviderId}/logout`
* End session endpoint: `http://localhost:9081/oidc/endpoint/{oidcProviderId}/end_session`

If the user’s OpenID Connect server SSO cookie is not available, such as when invoking the endpoints using a curl command, then an ID token belonging to the user will need to be appended to the request by using the `id_token_hint` query parameter.

* Logout endpoint with ID token hint: `http://localhost:9081/oidc/endpoint/{oidcProviderId}/logout?id_token_hint={id_token}`
* End session endpoint with ID token hint: `http://localhost:9081/oidc/endpoint/{oidcProviderId}/end_session?id_token_hint={id_token}`

Additionally, if the OpenID Connect server is configured with a SAML Identity Provider (IdP) by using link:https://openliberty.io/docs/latest/reference/feature/samlWeb-2.0.html[SAML Web Single Sign-On], then a logout at the IdP will also trigger the OpenID Connect server to send back-channel logout requests to the configured OpenID Connect clients.
```

Reply:
looks good, updated